pages/radar.md

```diff
@@ -7,6 +7,9 @@ permalink: /radar.html
 
 [/home](/)
 
+## [JSON Schema bundling finally formalized](https://json-schema.org/blog/posts/bundling-json-schema-compound-documents)
+8-September-2021: According [Unix principles](https://en.wikipedia.org/wiki/Unix_philosophy), you have to "write programs that do one thing but do it well". Smaller programs are much easier to maintain and test. The final cost of developing a large system built from small but reliable components with little cohesion is much less than the cost of a monolithic program with the same functionality - due to the price overhead for finding and fixing bugs. Moreover, small components can be written in different programming languages (the most suitable for each task). However, in this case, there is an issue of transferring the results from one component to another. The second postulate of the Unix philosophy states that programs should handle text strings as a universal interface. Complex data is usually transported using some kind of markup format. XML is often used, because it is human-readable, and can be automatically validated using XML Schema. However, XML has some [disadvantages](https://beginnersbook.com/2018/10/advantages-and-disadvantages-of-xml), such as a very [large overhead](https://www.xml.com/pub/a/2004/12/15/deviant.html) on the description of the data structure, which in some cases takes up more than the data itself. The JSON structure concentrates on the data, lowering the overhead. The way JSON self-describes, makes it the best performing data format to transport heterogenic data. We believe that the development of a mechanism for automatic JSON [validation](https://www.jsonschemavalidator.net/) based on the [schema](http://json-schema.org/understanding-json-schema/index.html) will make this format in the future a more popular mechanism for interprocess communication than XML.
```

Comment:
8-September-2021: According [Unix principles](https://en.wikipedia.org/wiki/Unix_philosophy), you have to "write programs that do one thing but do it well". Smaller programs are much easier to maintain and test. The final cost of developing a large system built from small but reliable components with little cohesion is much less than the cost of a monolithic program with the same functionality - due to the price overhead for finding and fixing bugs. Moreover, small components can be written in different programming languages (the most suitable for each task). However, there is an issue with transferring the results from one component to another in this case. The second postulate of the Unix philosophy states that programs should handle text strings as a universal interface. Complex data is usually transported using some type of markup format. XML is often used because it is human-readable and can be automatically validated using XML Schema. However, XML has some [disadvantages](https://beginnersbook.com/2018/10/advantages-and-disadvantages-of-xml), such as a very [large overhead](https://www.xml.com/pub/a/2004/12/15/deviant.html) on the description of the data structure, which in some cases takes up more than the data itself. The JSON structure concentrates on the data, lowering the overhead. The way JSON self-describes makes it the best performing data format to transport heterogenic data. We believe that the development of a mechanism for automatic JSON [validation](https://www.jsonschemavalidator.net/) built on the [schema](http://json-schema.org/understanding-json-schema/index.html) will make this format a more popular mechanism for interprocess communication than XML in the future.